sklearn/preprocessing/tests/test_common.py

```diff
@@ -45,8 +52,19 @@ def _get_valid_samples_by_column(X, col):
         (RobustScaler(with_centering=False), robust_scale, True, False, []),
     ],
 )
+@pytest.mark.parametrize(
+    "sparse_container",
+    [
+        CSR_CONTAINERS,
+        CSC_CONTAINERS,
+        COO_CONTAINERS,
+        LIL_CONTAINERS,
+        DOK_CONTAINERS,
+        BSR_CONTAINERS,
+    ],
+)
```

Comment:
```suggestion
@pytest.mark.parametrize(
    "sparse_container",
        CSR_CONTAINERS
        + CSC_CONTAINERS
        + COO_CONTAINERS
        + LIL_CONTAINERS
        + DOK_CONTAINERS
        + BSR_CONTAINERS
    ,
)
```
The code might be iterating through a list-of-lists instead of a single list combining all of them as you probably intended, how about trying this instead?

Reply:
Nope, it's even failing to pass internal test cases.

Reply:
> Nope, it's even failing to pass internal test cases.

What do you mean? What @bharatr21 suggests looks correct to me, but then you could get rid of the `for sparse_constructor in sparse_container:` loop and directly do:


```python
X_train_sp = sparse_container(X_train)
```